component/client-handler/src/test/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/PrivateKeyJWTClientAuthenticatorTest.java

```diff
@@ -107,6 +107,22 @@ public void testcanAuthenticate() throws IdentityOAuth2Exception {
         boolean received = privateKeyJWTClientAuthenticator.canAuthenticate(httpServletRequest, bodyContent,
                 oAuthClientAuthnContext);
         assertEquals(received, true, "A valid request refused to authenticate.");
+    }
+
+    @Test
+    public void testPrivateKeyJWTFlagAdded() {
```

Comment:
The reason to assert it in the exception handling section is because when the isValidAssertion method finds any mismatch during a parameter validation it does not return false but instead it throws an exception. There is no way to get the isValidAssertion method to return either true or false due to the private methods that are being used which cannot be mocked. That is the reason why there is not test written for the authenticateClient method in the PrivateKeyJWTClientAuthenticatorTests but instead the test is written in the JWTValidator tests. 

Basically in this test what we need to check is that even though the authenticateClient method fails to authenticate the request or not still this flag needs to be added because it is a request that should be authenticated via the PrivateKeyJWT client authenticator which was confirmed by the canAuthenticate method. 

Anyway to address the concern that you mentioned I have asserted the param in the try block in as well in 159511f58156995bb4ad0e05d019c4349ee04bcd so that by any chance in future if these validations were changed to return false instead of an exception, still it will not impact this unit test.